execution-core/src/transfer/phoenix.rs

```diff
@@ -418,6 +418,15 @@ impl Transaction {
         }
     }
 
+    /// Returns the memo used with the transaction, if any.
+    #[must_use]
+    pub fn memo(&self) -> Option<[u8; 512]> {
+        match self.exec()? {
+            ContractExec::Memo(memo) => Some(*memo),
+            _ => None,
+        }
+    }
+
```

Comment:
Not sure we need it for Phoenix too

Reply:
Perhaps not, but it doesn't hurt and makes the implementation simpler. If we want to remove this we would have to have separate enums for both.